app/src/main/java/org/mozilla/focus/session/ui/SessionsSheetFragment.kt

```diff
@@ -0,0 +1,138 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.session.ui
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewAnimationUtils
+import android.view.ViewGroup
+import android.view.animation.AccelerateInterpolator
+
+import org.mozilla.focus.R
+import org.mozilla.focus.activity.MainActivity
+import org.mozilla.focus.locale.LocaleAwareFragment
+import org.mozilla.focus.session.SessionManager
+import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.utils.OneShotOnPreDrawListener
+
+class SessionsSheetFragment : LocaleAwareFragment(), View.OnClickListener {
+
+    private var backgroundView: View? = null
+    private var cardView: View? = null
+    private var isAnimating: Boolean = false
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_sessionssheet, container, false)
+
+        backgroundView = view.findViewById(R.id.background)
+        backgroundView!!.setOnClickListener(this)
+
+        cardView = view.findViewById(R.id.card)
+        cardView!!.viewTreeObserver.addOnPreDrawListener(object : OneShotOnPreDrawListener(cardView!!) {
+            override fun onPreDraw(view: View) {
+                playAnimation(false)
+            }
+        })
+
+        val sessionsAdapter = SessionsAdapter(this)
+        SessionManager.getInstance().sessions.observe(this, sessionsAdapter)
+
+        val sessionView = view.findViewById<RecyclerView>(R.id.sessions)
+        sessionView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
+        sessionView.adapter = sessionsAdapter
+
+        return view
+    }
+
+    @Suppress("ComplexMethod")
+    private fun playAnimation(reverse: Boolean): Animator {
+        isAnimating = true
+
+        val offset = resources.getDimensionPixelSize(R.dimen.floating_action_button_size) / 2
+        val cx = cardView!!.measuredWidth - offset
```

Comment:
If the fragment becomes detached during the animation, won't this crash?

Reply:
I think it won't crash here as the `cardView` reference is never set to null - I made this more obviously by using `lateinit` in the updated patch. But I agree that this animation is pointlessly keeping a reference and "completing" the animation even though the fragment may no longer be attached - leaking the views temporarily.